CryptoPrices/Sources/Data/Sources/NetworkCore/APICoin.swift

```diff
@@ -0,0 +1,19 @@
+//
+//  APICoin.swift
+//  Data
+//
+//  Created by Doan Thieu on 01/12/2022.
+//
+
+import Foundation
+
+public struct APICoin: Decodable, Equatable {
```

Comment:
IMO in NetworkCore, we should define the core of configuration, API error ... or stuff like that, and APICoin is a model. Because of that, for clarifying, what do you think we should move model APICoin to its Model folder? 💭

Reply:
I think the only _things (objects)_ that belong to _Model_ are the ones we put in the `Entities` module of the `Domain` package. Because this is the center of the app, to make the app according to the required business logic many modules would depend on the `Entities` module, from `Data`, `Home`, and `MyCoin` to the app itself.

Speaking of `APICoin`, it is of course the _model_ in the sense that is a concrete type, but it is never used outside the context of the network layer (hence the `API` prefix). The only modules that depend on it are the ones that implement the network layer (`NetworkExtension` in this case).

You can see that as there's no Repository, UseCase and the app ever use this `APICoin`. They all work on the `Coin` abstraction defined in the `Entities`. They're free to define their own _model_ that conforms to the `Coin` interface. For e.g, the `Home` module might have `MyCoinUIModel` doing that. But in any case, the `APICoin` is never propagated up to a layer that's higher than the network.

That's why the `APICoin` belongs to the `NetworkCore`, and together with the `CoinAPIProtocol` to provide a blueprint, a specification of the network layer: `func myCoins() async throws -> [APICoin]`, etc, ...

Reply:
Thank you for the detailed explanation 🙇 . It's clear to me now.